File: libgomp/testsuite/libgomp.oacc-fortran/deep-copy-1.f90
! { dg-do run }

! Test of attach/detach with "acc data".

program dtype
  implicit none
  integer, parameter :: n = 512
  type mytype
    integer, allocatable :: a(:)
  end type mytype
  integer i

  type(mytype) :: var

  allocate(var%a(1:n))

!$acc data copy(var)
!$acc data copy(var%a)

!$acc parallel loop
  do i = 1,n
    var%a(i) = i
  end do
!$acc end parallel loop

!$acc end data
!$acc end data

  do i = 1,n
    if (i .ne. var%a(i)) stop 1
  end do

  deallocate(var%a)

end program dtype

File: libgomp/testsuite/libgomp.oacc-c-c++-common/deep-copy-1.c
#include <stdlib.h>
#include <assert.h>

struct dc
{
  int a;
  int *b;
};

int
main ()
{
  int n = 100, i;
  struct dc v = { .a = 3, .b = (int *) malloc (sizeof (int) * n) };

#pragma acc parallel loop copy(v.a, v.b[:n])
  for (i = 0; i < n; i++)
    v.b[i] = v.a;

  for (i = 0; i < 10; i++)
    assert (v.b[i] == v.a);

  return 0;
}
